src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    internal class LinterAnalyzer : IBicepAnalyzer
+    {
+        public static string AnalyzerName => "Bicep Internal Linter";
+        public static readonly bool IsCliInvoked;
+
+        private static readonly IEnumerable<Type> RuleTypes;
+        private static readonly ImmutableArray<IBicepAnalyzerRule> RuleSet;
+
+        static LinterAnalyzer()
+        {
+            var entryAssembly = Assembly.GetEntryAssembly();
+            IsCliInvoked = entryAssembly.Location.EndsWith("bicep.dll", true, CultureInfo.CurrentCulture);
```

Comment:
Assuming we move to a non-static constructor (based on Marcin's comment - which I think is a good idea), I think it would be worth having `isCliInvoked` be an explicit constructor argument rather than trying to detect it - it feels like that would be more robust, and also gives you the ability to test both scenarios easily.

Reply:
IsCliInvoked is removed. Plan is to add a "host" string to the constructor to differentiate origin of request. Currently unknown what the need is for this.